api/views.py

```diff
@@ -1,10 +1,25 @@
-from api.serializers import UserSerializer
-from rest_framework import viewsets
+from api.models import Todo
+from api.serializers import TodoSerializer, UserSerializer
+from rest_framework import viewsets, permissions, renderers
 from django.contrib.auth.models import User
+from api.permissions import IsAuthorOrReadOnly
+from rest_framework.decorators import action
 
 
 class UserViewSet(viewsets.ReadOnlyModelViewSet):
 
     queryset = User.objects.all()
     serializer_class = UserSerializer
 
+
+class TodoViewSet(viewsets.ModelViewSet):
+   
+    queryset = Todo.objects.all()
+    serializer_class = TodoSerializer
+    permission_classes = [permissions.IsAuthenticatedOrReadOnly,
+                          IsAuthorOrReadOnly]
+
+    @action(detail=True, renderer_classes=[renderers.StaticHTMLRenderer])
+
```

Comment:
Indentation here. And on other parts of the code is bad. Learn to indent code, I do not want to repeat it every time I look at your MR...